File: app/src/main/java/com/leonov_dev/currencyconverter/StockFragment.java
package com.leonov_dev.currencyconverter;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;
import com.leonov_dev.currencyconverter.data.CurrencyReplacement;
import com.leonov_dev.currencyconverter.databinding.ActivityStockBinding;


import java.util.ArrayList;

/**
 * A simple {@link Fragment} subclass.
 */
public class StockFragment extends Fragment {

    public final String LOG_TAG = this.getClass().getSimpleName();

    private StockConverterViewModel mStockViewModel;

    private ActivityStockBinding mStockBinding;

    private CurrencyReplacementAdapter mCurrencyAdapter;


    public StockFragment() {
        // Required empty public constructor
    }

    //TODO Implement lifecycle?
    @Override
    public void onResume() {
        super.onResume();
        mStockViewModel.loadRemoteData();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.activity_stock, container, false);

        mStockBinding = ActivityStockBinding.inflate(inflater, container, false);

        mStockViewModel = MainActivity.obtainViewModel(getActivity());

        mStockBinding.setViewmodel(mStockViewModel);

        setHasOptionsMenu(true);

        return mStockBinding.getRoot();
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        setupListAdapter();
    }

    private void setupListAdapter() {
        ListView listView =  mStockBinding.list;

        mCurrencyAdapter = new CurrencyReplacementAdapter(
                new ArrayList<CurrencyReplacement>(0),
                mStockViewModel
        );
        listView.setAdapter(mCurrencyAdapter);
    }

}

File: app/src/main/java/com/leonov_dev/currencyconverter/ui/home/stock/StockFragment.kt
package com.leonov_dev.currencyconverter.ui.home.stock

import android.os.Bundle
import androidx.fragment.app.Fragment

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.leonov_dev.currencyconverter.R

import com.leonov_dev.currencyconverter.ui.home.MainActivity
import com.leonov_dev.currencyconverter.ui.home.StockConverterViewModel


import java.util.ArrayList

class StockFragment : Fragment() {

    private lateinit var stockViewModel: StockConverterViewModel
    private var currencyAdapter: CurrencyReplacementAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(true)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val rootView = inflater.inflate(R.layout.fragment_stock, container, false)
        stockViewModel = MainActivity.obtainViewModel(requireActivity())
        return rootView
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        currencyAdapter = CurrencyReplacementAdapter(ArrayList(0))
    }

    override fun onResume() {
        super.onResume()
        stockViewModel.loadRemoteData()
    }
}